### PR TITLE
Add view on code host for Bitbucket Cloud

### PR DESCRIPTION
**Pull request by artemruts**
### Description

While doing a walkthrough with a customer @emchap noticed that we can't navigate from Sourcegraph to Bitbucket Cloud.

### Changes

This PR:

* Adds `View on Bitbucket Cloud` button to let users navigate from Sourcegraph to Bitbucket Cloud
* Adds support for links with `range` or `position` for Bitbucket Cloud and Bitbucket Server

### Tested

Since I touched the `range` and `position` logic a bit, I tested on repositories from:
* GitHub
* GitLab
* Bitbucket Server (https://bitbucket.sgdev.org/)
* Bitbucket Cloud (https://bitbucket.org/)

### Notes

Screen Shot 2021-03-23 at 11 48 16 PM

1. @quinnkeast Customer shared that the Bitbucket icon looks a bit like `delete` button. What could we do about this? 🤔 
2. I didn't add the `compare`/`diff` link functionality yet but can make another ticket and add it for Bitbucket (I think .org supports it)

### CODE REVIEW

File: cmd/repo-updater/repoupdater/server.go [hunk 1]
```diff
@@ -681,6 +682,25 @@ func newRepoInfo(r *types.Repo) (*protocol.RepoInfo, error) {
 			Blob:   pathAppend(root, "/browse/{path}?at={rev}"),
 			Commit: pathAppend(root, "/commits/{commit}"),
 		}
+	case extsvc.TypeBitbucketCloud:
+		{
```
Comment (unknwon) [inline cmd/repo-updater/repoupdater/server.go:686]: Hmm? Is this bracket necessary?

**Comment by quinnkeast** [comment]
>@quinnkeast Customer shared that the Bitbucket icon looks a bit like delete button. What could we do about this? 🤔

Oh dang it does. 

We can't change their logo in any way. One option is to use the non-flat version as an SVG, but that's a departure from the patterns we're already using, and I'd want to hear @AlicjaSuska's thoughts on that. It'd look something like this:

CleanShot 2021-03-24 at 13 56 10@2x

Otherwise... I'm tempted to say if it looks like a delete icon, that's not really our problem to solve. 😕

File: cmd/repo-updater/repoupdater/server.go [hunk 2]
```diff
@@ -681,6 +682,25 @@ func newRepoInfo(r *types.Repo) (*protocol.RepoInfo, error) {
...
+			basePath := pathAppend(href, "/src/{rev}/{path}")
+
+			info.Links = &protocol.RepoLinks{
+				Root:   href,
+				Tree:   basePath,
+				Blob:   basePath,
+				Commit: pathAppend(href, "/commits/{commit}"),
+			}
+
+		}
```
Comment (unknwon) [inline cmd/repo-updater/repoupdater/server.go:703]:
```suggestion
		repo := r.Metadata.(*bitbucketcloud.Repo)
		href := repo.Links.HTML.Href

		if href == "" {
			break
		}

		basePath := pathAppend(href, "/src/{rev}/{path}")

		info.Links = &protocol.RepoLinks{
			Root:   href,
			Tree:   basePath,
			Blob:   basePath,
			Commit: pathAppend(href, "/commits/{commit}"),
		}
```

**Comment by emchap** [comment]
> Otherwise... I'm tempted to say if it looks like a delete icon, that's not really our problem to solve. 😕

Yeah, that's sort of been where the customers I've spoken to have landed—it's a stylistic choice on their part. Has there been any discussion of using colors? Even if it was just a blue somewhere in Atlassian's color family that might help. (That said, def not a blocker.)

Thank you for tackling this!

**Review by emchap — APPROVED** [review]

**Review by tjkandala — APPROVED** [review]
Do we have tests for the external URL construction logic? If not, should we?

**Review by unknwon — APPROVED** [review]

**Comment by tsenart** [comment]
Nice!!

**Comment by ryanslade** [comment]
@artemruts Is there a reason we didn't merge this?

**Comment by eseliger** [comment]
As part of #wg-delete-week I am going to close this PR. If it’s still needed, feel free to reopen and restore the branch.

**Pull request closed without merging**